docs/usecase.rst

```diff
@@ -37,14 +72,37 @@ compute/data/communications abstractions,
 quality of service guarantees,
 and performance optimizations.
 
+.. seealso:: :doc:`executor`
+
 Generic sequential task execution
 ---------------------------------
 
 Units of work (nodes) describe the inputs and expected outputs,
 and identify a module able to perform the operation.
 
+| 1. Receive a node of work to execute.
+| 2. Import a module implementing the operation.
+| 3. Initialize or restore checkpoint state.
```

Comment:
What software component is responsible for recording/tracking states and checkpoints? Are we assuming these are kept in memory?

Reply:
This is under active consideration. I will answer more completely in a follow-up soon to come...